src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/NotifyService.kt

```diff
@@ -116,6 +120,62 @@ class NotifyAppointmentService(
         event.actionPlanSession.sessionNumber,
       )
 
+      when (event.type) {
+        ActionPlanAppointmentEventType.ATTENDANCE_RECORDED -> {
+          emailSender.sendEmail(
+            appointmentNotAttendedTemplateID,
+            ppDetails.email,
+            mapOf(
+              "ppFirstName" to ppDetails.firstName,
+              "referenceNumber" to referral.referenceNumber!!,
+              "attendanceUrl" to location.toString(),
+            )
+          )
+        }
+        ActionPlanAppointmentEventType.BEHAVIOUR_RECORDED -> {
+          emailSender.sendEmail(
+            concerningBehaviourTemplateID,
+            ppDetails.email,
+            mapOf(
+              "ppFirstName" to ppDetails.firstName,
+              "referenceNumber" to referral.referenceNumber!!,
+              "sessionUrl" to location.toString(),
+            )
+          )
+        }
+      }
+    }
+  }
+}
+
+@Service
+class NotifyAppointmentService(
+  @Value("\${notify.templates.appointment-not-attended}") private val appointmentNotAttendedTemplateID: String,
+  @Value("\${notify.templates.concerning-behaviour}") private val concerningBehaviourTemplateID: String,
+  @Value("\${interventions-ui.baseurl}") private val interventionsUIBaseURL: String,
+  @Value("\${interventions-ui.locations.probation-practitioner.supplier-assessment-feedback}") private val ppSAASessionFeedbackLocation: String,
+  private val emailSender: EmailSender,
+  private val hmppsAuthService: HMPPSAuthService,
+) : ApplicationListener<AppointmentEvent>, NotifyService {
+  companion object : KLogging()
+  @AsyncEventExceptionHandling
+  override fun onApplicationEvent(event: AppointmentEvent) {
+    if (event.notifyPP) {
+      val referral = event.appointment.referral
+      val ppDetails = hmppsAuthService.getUserDetail(referral.getResponsibleProbationPractitioner())
+
+      val location = when (event.appointmentType) {
+        AppointmentType.SUPPLIER_ASSESSMENT -> generateResourceUrl(
+          interventionsUIBaseURL,
+          ppSAASessionFeedbackLocation,
+          event.appointment.referral.id
+        )
+        AppointmentType.SERVICE_DELIVERY -> run {
+          logger.error("action plan session should not be using the shared appointment service.")
+          return
```

Comment:
🙋‍♂️ Could this be a code smell? If we distinguish between `AppointmentEvent` and `ActionPlanAppointmentEvent` as distinct types, do we also need the `appointmentType`?